### PR TITLE
Completes the description of community provider's lifecycle

### PR DESCRIPTION
**Pull request by potiuk**
This PR separates lifecycle of providers to separate chapter, adds
release process and removal process description.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: PROVIDERS.rst [hunk 1]
```diff
@@ -251,3 +272,48 @@ The suspension may be triggered by any committer after the following criteria ar
...
+opinion on the removal.
+
+There are the following consequences (or lack of them) of removing the provider:
+
+* One last release of the provider is done with documentation updated informing that the provider is no
+  longer maintained by the Apache Airflow community - linking to this page. This information should also
+  find its way to the package documentation and consequently - to the description of the package in PyPI.
+* An ``[ANNOUNCE]`` thread is sent to the devlist and user list announcing removal of the provider
+* The released provider packages remain available on PyPI and in the Downloads/Archives of the Apache
+  Software Foundation, also it remains in the Index of the Apache Airflow Providers at
```
Comment (eladkal) [inline PROVIDERS.rst]: Do we keep it in svn?

Comment (potiuk) [reply PROVIDERS.rst]: Good point.

Indeed SVN finally gets to "Downloads/Archive": https://downloads.apache.org/airflow/ you can see latest release there and https://archive.apache.org/dist/airflow/  include all historical releases. 

We cannot remove things from archives (and that's fine), but indeed we can (and probably should) releave it from SVN which will remove it from "downloads".

File: PROVIDERS.rst [hunk 2]
```diff
@@ -251,3 +272,48 @@ The suspension may be triggered by any committer after the following criteria ar
...
+  `Airflow Documentation <https://airflow.apache.org/docs/>`_ with note ``(not maintained)`` next to it.
```
Comment (potiuk) [inline PROVIDERS.rst]:
```suggestion
* The released provider packages remain available on PyPI and in the
   `Archives <https://archive.apache.org/dist/airflow/providers/>`_ of the Apache
   Software Foundation, while they are removed from the 
   `Downlads <https://downloads.apache.org/airflow/providers/>`_ .
   Also it remains in the Index of the Apache Airflow Providers documentation at
   `Airflow Documentation <https://airflow.apache.org/docs/>`_ with note ``(not maintained)`` next to it.
```

File: PROVIDERS.rst (hunk 1 above)
Comment (potiuk) [reply PROVIDERS.rst]: Updated/Clarified.

File: PROVIDERS.rst [hunk 3]
```diff
@@ -251,3 +272,51 @@ The suspension may be triggered by any committer after the following criteria ar
...
+There are the following consequences (or lack of them) of removing the provider:
+
+* One last release of the provider is done with documentation updated informing that the provider is no
+  longer maintained by the Apache Airflow community - linking to this page. This information should also
+  find its way to the package documentation and consequently - to the description of the package in PyPI.
+* An ``[ANNOUNCE]`` thread is sent to the devlist and user list announcing removal of the provider
+* The released provider packages remain available on PyPI and in the
+   `Archives <https://archive.apache.org/dist/airflow/providers/>`_ of the Apache
+   Software Foundation, while they are removed from the 
+   `Downlads <https://downloads.apache.org/airflow/providers/>`_ .
```
Comment (hussein-awala) [inline PROVIDERS.rst]:
```suggestion
   `Downloads <https://downloads.apache.org/airflow/providers/>`_ .
```

**Review by hussein-awala — APPROVED** [review]
I like the proposed lifecycle!

File: PROVIDERS.rst [hunk 4]
```diff
@@ -251,3 +272,51 @@ The suspension may be triggered by any committer after the following criteria ar
...
+
+There are the following consequences (or lack of them) of removing the provider:
+
+* One last release of the provider is done with documentation updated informing that the provider is no
+  longer maintained by the Apache Airflow community - linking to this page. This information should also
+  find its way to the package documentation and consequently - to the description of the package in PyPI.
+* An ``[ANNOUNCE]`` thread is sent to the devlist and user list announcing removal of the provider
+* The released provider packages remain available on PyPI and in the
+   `Archives <https://archive.apache.org/dist/airflow/providers/>`_ of the Apache
+   Software Foundation, while they are removed from the
```
Comment (potiuk) [inline PROVIDERS.rst]:
```suggestion
   Software Foundation, while they are removed from the
```

File: PROVIDERS.rst [hunk 5]
```diff
@@ -251,3 +272,51 @@ The suspension may be triggered by any committer after the following criteria ar
...
+----------------------------
+
+The providers can be removed from main branch of Airflow when the community agrees that there should be no
+more updates to the providers done by the community - except maybe potentially security fixes found. There
+might be various reasons for the providers to be removed:
+
+* the service they connect to is no longer available
+* the dependencies for the provider are not maintained any more and there is no viable alternative
+* there is another, more popular provider that supersedes community provider
+* etc. etc.
```
Comment (Taragolis) [inline PROVIDERS.rst:286]: 😉 💯

**Comment by potiuk** [comment]
Looking for more comments - I will put it up for a vote in a day or two - I am looking forward to finally getting the full lifecycle of our providers described.

File: PROVIDERS.rst [hunk 6]
```diff
@@ -134,10 +152,13 @@ provider when we increase minimum Airflow version.
 Increasing the minimum version ot the Providers is one of the reasons why 3rd-party provider maintainers
 might want to maintain their own providers - as they can decide to support older versions of Airflow.
 
-3rd-party providers
-===================
+3rd-parties relation to community providers
+-------------------------------------------
 
-Providers, can (and it is recommended for 3rd-party providers) also be maintained and releases by 3rd parties.
+Providers, can (and it is recommended for 3rd-party services) also be maintained and releases by 3rd parties,
```
Comment (pankajkoti) [inline PROVIDERS.rst]:
```suggestion
Providers, can (and it is recommended for 3rd-party services) also be maintained and released by 3rd parties,
```

File: PROVIDERS.rst [hunk 7]
```diff
@@ -109,8 +117,18 @@ the community. Also it is often easier to advertise and promote usage of the pro
...
 
-Minimum supported version of Airflow for Community managed providers
-====================================================================
+Community providers release process
+-----------------------------------
+
+The community providers are released regularly (usually every 2 weeks) in batches consisting of any providers
+that need to be released because they changed since last release. The release manager decides which providers
+to include and whether some or all providers should be released (see the next chapter about upgrading the
+minimum version of Airflow for example of case where we release all active, not suspended providers,
```
Comment (pankajkoti) [inline PROVIDERS.rst]:
```suggestion
minimum version of Airflow for example the case where we release all active meaning non-suspended providers,
```

File: PROVIDERS.rst [hunk 8]
```diff
@@ -134,10 +152,13 @@ provider when we increase minimum Airflow version.
...
-3rd-party providers
-===================
+3rd-parties relation to community providers
+-------------------------------------------
 
-Providers, can (and it is recommended for 3rd-party providers) also be maintained and releases by 3rd parties.
+Providers, can (and it is recommended for 3rd-party services) also be maintained and releases by 3rd parties,
+but for multiple reasons we might decide to keep those providers as community managed providers - mostly
+due to prevalence and popularity of the 3rd-party services and use cases they serve among our community. There
+are however certain condition and expectations we have in order
```
Comment (pankajkoti) [inline PROVIDERS.rst]:
```suggestion
are however certain conditions and expectations we have in order.
```

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**